library/core/src/prelude/v1.rs

```diff
@@ -69,29 +68,26 @@ pub use crate::{
 #[doc(no_inline)]
 pub use crate::concat_bytes;
 
+// Do not `doc(inline)` these `doc(hidden)` items.
```

Comment:
Does this comment make sense?
The items are hidden anyway, so it doesn't make any difference whether we inline them or not?
(Especially considering that `doc(inline)` is the default in this situation.)

Reply:
> `doc(inline)` is the default in this situation

So, while this is currently true, I have to admit having chosen not to use `doc(inline)` explicitly since there is a change I've been considering to suggest to the `rustdoc` team, which would change, at the very least, the same-crate re-export of a `#[doc(hidden)]` item _when not using `#[doc(inline)]`_.

For a crate wanting to feature `macro_export`-ed macros scoped within a module, currently, there are three options:

  - ```rust
    pub mod module {
        #[macro_export]
        macro_rules! __m__ {() => ()}
        pub use __m__ as m;
    }
    ```

    Drawback: `__m__!` is visible at the crate root.

      - Workaround: use js hacks to hide it in the aftermath ([an ugly example](https://docs.rs/uninit/latest/src/uninit/read/mod.rs.html#173))

         Drawback: so hacky and ugly.

  - Another solution: use a helper crate that performs the `#[macro_export]`s, and then `pub use` them in a scoped fashion from the frontend crate ([an example](https://docs.rs/konst_macro_rules)).

    Drawback: on top of being more cumbersome, it breaks `$crate`.

The best solution for these things that occurred to me would be for `#[doc(inline)]` to override `#[doc(hidden)]` (or have a `doc(inline_unhidden)` for this, but that seems to be way more work). Indeed, `doc(inline)` expresses an intent to have the item be documented _in situ_, so it makes sense to have it express `_unhidden` semantics as well, for the rare case where it could matter.

And this would be one such case, where we'd thus have not to use `doc(inline)`. I thus find it less error-prone to do it this way, if that makes sense?

Reply:
Ok, I don't want to dig too much into this.